game.py

```diff
@@ -223,17 +196,15 @@ def perform_self_destruct(self, coords: CoordPair) -> Tuple[bool, str]:
     def perform_move(self, coords: CoordPair) -> Tuple[bool, str]:
         """Validate and perform a move expressed as a CoordPair. TODO: WRITE MISSING CODE!!!"""
         if self.is_valid_move(coords):
-            if self.is_movement(coords):
+            if self.is_empty(coords.dst):
                 return self.perform_movement(coords)
-            elif self.is_attack(coords):
+            elif self.next_player != self.get(coords.dst).player:
```

Comment:
also for all of these we should leave them as separate functions is_attack or is_move and have it return a boolean it makes it easier to read and the tiny amount of optimization we are getting from this is not worth making the code less readable